elements/lisk-p2p/src/peer_pool.ts

```diff
@@ -96,6 +96,8 @@ interface PeerPoolConfig {
 	readonly maxOutboundConnections: number;
 	readonly maxInboundConnections: number;
 	readonly outboundShuffleInterval?: number;
+	readonly evictionProtectionEnabled?: boolean;
+	readonly evictionProtectionRatio?: number;
```

Comment:
I think that we should make `peerBanTime`, `outboundShuffleInterval`, `evictionProtectionEnabled` and `evictionProtectionRatio` compulsory because otherwise we have to do a lot of type casting like `this.someProperty as number` in our code and we lose type safety.

We can keep `ackTimeout` and `connectTimeout` as optional because those are passed to SocketCluster so we don't need to worry about type casting them.